### PR TITLE
closer to jdk-17 changes

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -209,6 +209,8 @@
 			<plugin>
 				<groupId>io.spring.javaformat</groupId>
 				<artifactId>spring-javaformat-maven-plugin</artifactId>
+				<!-- Remove version when : https://github.com/spring-cloud/spring-cloud-build/pull/235 is integrated and released -->
```
Comment (wind57) [inline pom.xml:212]: I will take care of this one also, when it is fixed over there

File: spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigReloadUtil.java [hunk 2]
```diff
@@ -108,14 +108,16 @@ static List<MapPropertySource> locateMapPropertySources(PropertySourceLocator pr
...
-			result.add((MapPropertySource) propertySource);
+		if (propertySource instanceof MapPropertySource mapPropertySource) {
+			result.add(mapPropertySource);
 		}
 		else if (propertySource instanceof CompositePropertySource source) {
 
-			List<MapPropertySource> list = source.getPropertySources().stream()
-					.filter(p -> p instanceof MapPropertySource).map(x -> (MapPropertySource) x).toList();
-			result.addAll(list);
+			source.getPropertySources().forEach(x -> {
```
Comment (wind57) [inline spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigReloadUtil.java:116]: this is so much easier to read, I am really sorry it did not occur to me when I was here in this code path before

File: src/checkstyle/checkstyle-suppressions.xml [hunk 3]
```diff
@@ -15,6 +15,5 @@
 	<suppress files=".*ConfigurationWatcherApplication\.java" checks="HideUtilityClassConstructor"/>
 	<suppress files=".*DiscoveryServerApplication\.java" checks="HideUtilityClassConstructor"/>
 	<suppress files=".*KubernetesConfigServerApplication\.java" checks="HideUtilityClassConstructor"/>
-	<suppress files=".*NormalizedSource\.java" checks="WhitespaceAfter" />
```
Comment (wind57) [inline src/checkstyle/checkstyle-suppressions.xml:18]: if we upgrade formatter plugin, we do not require this one anymore

**Comment by wind57** [comment]
@ryanjbaxter  let's see if I get lucky :) one more to look at please? :)

File: pom.xml (hunk 1 above)
Comment (ryanjbaxter) [reply pom.xml:212]: Let me know when the PR is ready and we can do this all at once...or you can separate this change out into a separate PR

Comment (wind57) [reply pom.xml:212]: Right, this is ready. Let me know when you close and release that one, and Ill integrate it in after after that. I hope I understood you correctly

Comment (wind57) [reply pom.xml:212]: @ryanjbaxter build has passed over there, FYI

Comment (ryanjbaxter) [reply pom.xml:212]: Its merged, the build should only take a few minutes before snapshots are published

**Comment by wind57** [comment]
@ryanjbaxter ready.

**Pull request merged**